<a href="https://colab.research.google.com/github/MOR1968/Exploratory-Data-Analysis/blob/main/analysexploradonnee.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [4]:
filepath='https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DA0101EN-SkillsNetwork/labs/Data%20files/automobileEDA.csv'
df = pd.read_csv(filepath, header=0)

In [5]:
df.head()

symboling  normalized-losses         make aspiration num-of-doors  \
0          3                122  alfa-romero        std          two   
1          3                122  alfa-romero        std          two   
2          1                122  alfa-romero        std          two   
3          2                164         audi        std         four   
4          2                164         audi        std         four   

    body-style drive-wheels engine-location  wheel-base    length  ...  \
0  convertible          rwd           front        88.6  0.811148  ...   
1  convertible          rwd           front        88.6  0.811148  ...   
2    hatchback          rwd           front        94.5  0.822681  ...   
3        sedan          fwd           front        99.8  0.848630  ...   
4        sedan          4wd           front        99.4  0.848630  ...   

   compression-ratio  horsepower  peak-rpm city-mpg highway-mpg    price  \
0                9.0       111.0    5000.0       21          27  13495.0   
1                9.0       111.0    5000.0       21          27  16500.0   
2                9.0       154.0    5000.0       19          26  16500.0   
3               10.0       102.0    5500.0       24          30  13950.0   
4                8.0       115.0    5500.0       18          22  17450.0   

  city-L/100km  horsepower-binned  diesel  gas  
0    11.190476             Medium       0    1  
1    11.190476             Medium       0    1  
2    12.368421             Medium       0    1  
3     9.791667             Medium       0    1  
4    13.055556             Medium       0    1  

[5 rows x 29 columns]

###  Verification des Types de variable
il est important de comprendre d’abord de quel type de variable il s’agit. Cela nous aidera à trouver la méthode de visualisation la plus appropriée pour cette variable.

In [7]:
# liste les types de variables pour chaque colonne
print(df.dtypes)

symboling              int64
normalized-losses      int64
make                  object
aspiration            object
num-of-doors          object
body-style            object
drive-wheels          object
engine-location       object
wheel-base           float64
length               float64
width                float64
height               float64
curb-weight            int64
engine-type           object
num-of-cylinders      object
engine-size            int64
fuel-system           object
bore                 float64
stroke               float64
compression-ratio    float64
horsepower           float64
peak-rpm             float64
city-mpg               int64
highway-mpg            int64
price                float64
city-L/100km         float64
horsepower-binned     object
diesel                 int64
gas                    int64
dtype: object


In [9]:
df['peak-rpm'].dtypes

dtype('float64')

Par exemple nous pouvons calculer la correlation entre les variables de types int64 et float64 en utilisant la methode corr()

In [10]:
# Selectionnons seulement les variables numeriques
numeric_df = df.select_dtypes(include=['float64', 'int64'])
numeric_df.corr()

symboling  normalized-losses  wheel-base    length  \
symboling           1.000000           0.466264   -0.535987 -0.365404   
normalized-losses   0.466264           1.000000   -0.056661  0.019424   
wheel-base         -0.535987          -0.056661    1.000000  0.876024   
length             -0.365404           0.019424    0.876024  1.000000   
width              -0.242423           0.086802    0.814507  0.857170   
height             -0.550160          -0.373737    0.590742  0.492063   
curb-weight        -0.233118           0.099404    0.782097  0.880665   
engine-size        -0.110581           0.112360    0.572027  0.685025   
bore               -0.140019          -0.029862    0.493244  0.608971   
stroke             -0.008245           0.055563    0.158502  0.124139   
compression-ratio  -0.182196          -0.114713    0.250313  0.159733   
horsepower          0.075819           0.217299    0.371147  0.579821   
peak-rpm            0.279740           0.239543   -0.360305 -0.285970   
city-mpg           -0.035527          -0.225016   -0.470606 -0.665192   
highway-mpg         0.036233          -0.181877   -0.543304 -0.698142   
price              -0.082391           0.133999    0.584642  0.690628   
city-L/100km        0.066171           0.238567    0.476153  0.657373   
diesel             -0.196735          -0.101546    0.307237  0.211187   
gas                 0.196735           0.101546   -0.307237 -0.211187   

                      width    height  curb-weight  engine-size      bore  \
symboling         -0.242423 -0.550160    -0.233118    -0.110581 -0.140019   
normalized-losses  0.086802 -0.373737     0.099404     0.112360 -0.029862   
wheel-base         0.814507  0.590742     0.782097     0.572027  0.493244   
length             0.857170  0.492063     0.880665     0.685025  0.608971   
width              1.000000  0.306002     0.866201     0.729436  0.544885   
height             0.306002  1.000000     0.307581     0.074694  0.180449   
curb-weight        0.866201  0.307581     1.000000     0.849072  0.644060   
engine-size        0.729436  0.074694     0.849072     1.000000  0.572609   
bore               0.544885  0.180449     0.644060     0.572609  1.000000   
stroke             0.188829 -0.062704     0.167562     0.209523 -0.055390   
compression-ratio  0.189867  0.259737     0.156433     0.028889  0.001263   
horsepower         0.615077 -0.087027     0.757976     0.822676  0.566936   
peak-rpm          -0.245800 -0.309974    -0.279361    -0.256733 -0.267392   
city-mpg          -0.633531 -0.049800    -0.749543    -0.650546 -0.582027   
highway-mpg       -0.680635 -0.104812    -0.794889    -0.679571 -0.591309   
price              0.751265  0.135486     0.834415     0.872335  0.543155   
city-L/100km       0.673363  0.003811     0.785353     0.745059  0.554610   
diesel             0.244356  0.281578     0.221046     0.070779  0.054458   
gas               -0.244356 -0.281578    -0.221046    -0.070779 -0.054458   

                     stroke  compression-ratio  horsepower  peak-rpm  \
symboling         -0.008245          -0.182196    0.075819  0.279740   
normalized-losses  0.055563          -0.114713    0.217299  0.239543   
wheel-base         0.158502           0.250313    0.371147 -0.360305   
length             0.124139           0.159733    0.579821 -0.285970   
width              0.188829           0.189867    0.615077 -0.245800   
height            -0.062704           0.259737   -0.087027 -0.309974   
curb-weight        0.167562           0.156433    0.757976 -0.279361   
engine-size        0.209523           0.028889    0.822676 -0.256733   
bore              -0.055390           0.001263    0.566936 -0.267392   
stroke             1.000000           0.187923    0.098462 -0.065713   
compression-ratio  0.187923           1.000000   -0.214514 -0.435780   
horsepower         0.098462          -0.214514    1.000000  0.107885   
peak-rpm          -0.065713          -0.435780    0.107885  1.000000   
city-mpg 

Trouvez la corrélation entre les colonnes suivantes : bore, stroke, compression-ratio et horsepower.

Astuce : si vous souhaitez sélectionner ces colonnes, utilisez la syntaxe suivante :
df[['bore','stroke','compression-ratio','horsepower']]

In [12]:
df[['bore','stroke','compression-ratio','horsepower']].corr()

bore    stroke  compression-ratio  horsepower
bore               1.000000 -0.055390           0.001263    0.566936
stroke            -0.055390  1.000000           0.187923    0.098462
compression-ratio  0.001263  0.187923           1.000000   -0.214514
horsepower         0.566936  0.098462          -0.214514    1.000000

**bore et stroke** : La corrélation est de -0.055390, ce qui est très proche de 0. Cela indique une très faible corrélation linéaire entre le bore et le stroke.

**bore et compression-ratio** : La corrélation est de 0.001263, également très proche de 0. Cela indique une très faible corrélation linéaire entre le bore et le taux de compression.

**bore et horsepower** : La corrélation est de 0.566936. C'est une corrélation modérée positive. Cela suggère qu'à mesure que le bore augmente, la puissance (horsepower) a tendance à augmenter également.

***stroke et compression-ratio ***: La corrélation est de 0.187923, ce qui est une corrélation faible positive.

**stroke et horsepower** : La corrélation est de 0.098462, ce qui est une corrélation très faible positive.
compression-ratio et horsepower : La corrélation est de -0.214514. C'est une

**corrélation faible négative**. Cela suggère qu'à mesure que le taux de compression augmente, la puissance (horsepower) a tendance à légèrement diminuer.